src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -4256,7 +4261,66 @@ private void GenerateOneCode()
                     //:         break Backward;
                     {
                         string str = _strings![Operand(0)];
+                        Label multiCode = DefineLabel();
+                        if (!IsRightToLeft())
+                        {
+                            // if (runtextend - runtextpos < c)
+                            Ldloc(_runtextendLocal!);
+                            Ldloc(_runtextposLocal!);
+                            Sub();
+                            Ldc(str.Length);
+                            BgeFar(multiCode);
+                            // if (!_caseInsensitive && _maxBacktrackPosition != -1 && runtextpos > _maxBacktrackPosition)
+                            if (!IsCaseInsensitive())
+                            {
+                                Ldloc(_maxBacktrackPositionLocal!);
+                                Ldc(-1);
+                                BeqFar(_backtrack);
+                                Ldloc(_runtextposLocal!);
+                                Ldloc(_maxBacktrackPositionLocal!);
+                                BleFar(_backtrack);
+                                // runtextpos = _maxBacktrackPosition;
+                                Ldloc(_maxBacktrackPositionLocal!);
+                                Stloc(_runtextposLocal!);
+                                // ReadOnlySpan<char> runtextSpan = runtext.AsSpan(_maxBacktrackPosition, runtextend - _maxBacktractPosition);
+                                Ldloc(_runtextLocal!);
+                                Ldloc(_maxBacktrackPositionLocal!);
+                                Ldloc(_runtextendLocal!);
+                                Ldloc(_maxBacktrackPositionLocal!);
+                                Sub();
+                                using (RentedLocalBuilder textSpanLocal = RentReadOnlySpanCharLocal())
```

Comment:
Nit: textSpanLocal => runtextSpanLocal to match the name you're using in the comments